### PR TITLE
New ScheduleOn APIs

### PR DESCRIPTION
**Pull request by duke8253**
Some new ScheduleOn APIs, presented during 2018 fall summit (https://cwiki.apache.org/confluence/display/TS/Presentations+-+2018?preview=/75958509/94798265/2018%20fall%20summit.pptx). 

```
TSEventThread TSEventThreadSelf(void);
TSAction TSContSchedule(TSCont contp, TSHRTime timeout);
TSAction TSContScheduleOnPool(TSCont contp, TSHRTime timeout, TSThreadPool tp);
TSAction TSContScheduleOnThread(TSCont contp, TSHRTime timeout, TSEventThread ethread);
TSAction TSContScheduleEvery(TSCont contp, TSHRTime every /* millisecs */);
TSAction TSContScheduleEveryOnPool(TSCont contp, TSHRTime every /* millisecs */, TSThreadPool tp);
TSAction TSContScheduleEveryOnThread(TSCont contp, TSHRTime every /* millisecs */, TSEventThread ethread);
TSReturnCode TSContThreadAffinitySet(TSCont contp, TSEventThread ethread);
TSEventThread TSContThreadAffinityGet(TSCont contp);
void TSContThreadAffinityClear(TSCont contp);
```

Note: the behavior of the original `TSContSchedule` is now `TSContScheduleOnPool`.

Here's a link to some previous discussions: https://github.com/duke8253/trafficserver/pull/5

### CODE REVIEW

**Comment by duke8253** [comment]
[approve ci autest]

File: src/traffic_server/InkAPI.cc [hunk 1]
```diff
@@ -4430,36 +4458,89 @@ TSContSchedule(TSCont contp, ink_hrtime timeout, TSThreadPool tp)
     break;
```
[the remark this answers is not in the document]
Comment (duke8253) [reply src/traffic_server/InkAPI.cc]: not quite sure what you mean by handle? like abort?

**Comment by duke8253** [comment]
[approve ci autest]

File: src/traffic_server/InkAPI.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (zizhong) [reply src/traffic_server/InkAPI.cc]: Oh, it already got a default: branch. Never mind.

File: doc/developer-guide/api/functions/TSContScheduleOnPool.en.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+========
+
+`#include <ts/ts.h>`
+
+.. function:: TSAction TSContScheduleOnPool(TSCont contp, TSHRTime timeout, TSThreadPool tp)
+
+Description
+===========
+
+Mostly same as :func:`TSContSchedule`. Schedules :arg:`contp` on a random thread that belongs to :arg:`tp`.
```
Comment (ywkaras) [inline doc/developer-guide/api/functions/TSContScheduleOnPool.en.rst]: Mostly the same

**Comment by duke8253** [comment]
I'm trying to add a new lifecycle hook to make the tests better, sent out on the mailing list, will focus on that first.

**Pull request merged**